python/lsst/atmospec/processStar.py

```diff
@@ -509,11 +513,17 @@ def setDefaults(self):
     def validate(self):
         super().validate()
         uvspecPath = shutil.which('uvspec')
-        if uvspecPath is None and self.doFitAtmosphere is True:
-            raise FieldValidationError(self.__class__.doFitAtmosphere, self, "uvspec is not in the path,"
+        try:
+            import getObsAtmo
+        except ModuleNotFoundError:
+            getObsAtmo = None
+        if uvspecPath is None and getObsAtmo is None and self.doFitAtmosphere is True:
+            raise FieldValidationError(self.__class__.doFitAtmosphere, self,
+                                       "uvspec is not in the path nor getObsAtmo is instaled,"
                                        " but doFitAtmosphere is True.")
-        if uvspecPath is None and self.doFitAtmosphereOnSpectrogram is True:
-            raise FieldValidationError(self.__class__.doFitAtmosphereOnSpectrogram, self, "uvspec is not in"
+        if uvspecPath is None and getObsAtmo is None and self.doFitAtmosphereOnSpectrogram is True:
+            raise FieldValidationError(self.__class__.doFitAtmosphereOnSpectrogram, self,
+                                       "uvspec is not in the path nor getObsAtmo is installed,"
```

Comment:
I don't think this logic is thorough enough - you could still have one of these not be `None`, but be asking to use the one which _is_ `None`, so this doesn't protect against that. Can you rewrite this to make sure that whatever option is selected will actually be available when it's time to use it? (This validate early was the compromise we found that meant we didn't have to add these awkward packages to the env, but the downside to that is the pipeline _must_ be able to run the way it's been asked to).